pkg/controllers/capr/managesystemagent/managesystemagent.go

```diff
@@ -208,6 +213,16 @@ func installer(cluster *rancherv1.Cluster, secretName string) []runtime.Object {
 		})
 	}
 
+	controlPlane, err := h.rkeControlPlane.Get(cluster.Namespace, cluster.Name, metav1.GetOptions{})
```

Comment:
Instead of retrieving the controlPlane, could we use the value on the v1 Cluster object at `.Spec.RKEConfig.DataDirectories.SystemAgent`?  
If that works, we can simplify the code a little bit.

Reply:
Couldn't do it exactly but was able to tweak the `GetSystemAgentDataDir` function to work with either a controlplane or provisioning cluster object.